commit 5f2e584be1fa2ae417abdf6841a00efae94d0a7e
Author: Kirill A. Korinsky <kirill@korins.ky>
Date:   Mon Nov 29 23:21:25 2021 +0100

    fixed build on old macOS
    
    It allows to build on macOS up to 10.10.
    
    Anyway, it is uses `NSOpenGLContext.pixelFormat` which was added at
    10.10 and removing it seems like as a huge and meanless job.

diff --git a/OgreMain/src/OSX/OgreFileSystemLayer.cpp b/OgreMain/src/OSX/OgreFileSystemLayer.cpp
index 2fd372d09..16bc152bd 100644
--- a/OgreMain/src/OSX/OgreFileSystemLayer.cpp
+++ b/OgreMain/src/OSX/OgreFileSystemLayer.cpp
@@ -30,6 +30,8 @@
 #include <pwd.h>
 #include <dlfcn.h>
 #include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
 
 namespace Ogre
 {
diff --git a/RenderSystems/GLSupport/src/OSX/OgreOSXCocoaWindow.mm b/RenderSystems/GLSupport/src/OSX/OgreOSXCocoaWindow.mm
index 6a220f51b..6cab88362 100644
--- a/RenderSystems/GLSupport/src/OSX/OgreOSXCocoaWindow.mm
+++ b/RenderSystems/GLSupport/src/OSX/OgreOSXCocoaWindow.mm
@@ -33,6 +33,7 @@ THE SOFTWARE.
 
 #import "OgreGLRenderSystemCommon.h"
 #import "OgreGLNativeSupport.h"
+#import <AppKit/AppKit.h>
 #import <AppKit/NSScreen.h>
 #import <AppKit/NSOpenGLView.h>
 #import <QuartzCore/CVDisplayLink.h>
